### PR TITLE
buildFHSUserEnv: extend PATH and LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by abbradar**
This allows one to run applications from PATH and override libraries for
applications inside chrootenv.

###### Motivation for this change

Useful for development environments.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Tested by running Steam and my local development environments.

### CODE REVIEW

**Comment by abbradar** [comment]
I'll merge it in several days if there are no objections.

**Comment by Mic92** [comment]
I think it make sense to match the behavior of `nix-shell` here.

**Comment by abbradar** [comment]
@Mic92 What exactly do you mean?

**Comment by Mic92** [comment]
I mean that a `nix-shell` environment would also keep `PATH` unmodified unless the `--pure` flag is used.

**Comment by abbradar** [comment]
Ah, yes. No `--pure` flag for now but buildFHSUserEnv isn't generally suited for isolation. 

On May 25, 2019 12:01:40 PM GMT+03:00, "Jörg Thalheim" <notifications@github.com> wrote:
>I mean that a `nix-shell` environment would also keep `PATH` unmodified
>unless the `--pure` flag is used.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/62040#issuecomment-495890636

-- 
Nikolay.

**Review by primeos — APPROVED** [review]
LGTM, this would e.g. also make #61839 possible.

If it should be configurable we could introduce an attribute like `pure ? false` that can be passed to `env.nix` via `buildFHSUserEnv` (@Mic92 would that be enough?).

(Or we could do something like `nix-shell` does and store the original `PATH` to another environment variable like `HOST_PATH`.)

File: pkgs/build-support/build-fhs-userenv/env.nix [hunk 1]
```diff
@@ -52,8 +52,8 @@ let
   etcProfile = writeText "profile" ''
     export PS1='${name}-chrootenv:\u@\h:\w\$ '
     export LOCALE_ARCHIVE='/usr/lib/locale/locale-archive'
-    export LD_LIBRARY_PATH='/run/opengl-driver/lib:/run/opengl-driver-32/lib:/usr/lib:/usr/lib32'
-    export PATH='/run/wrappers/bin:/usr/bin:/usr/sbin'
+    export LD_LIBRARY_PATH="/run/opengl-driver/lib:/run/opengl-driver-32/lib:/usr/lib:/usr/lib32:$LD_LIBRARY_PATH"
```
Comment (memberbetty) [inline pkgs/build-support/build-fhs-userenv/env.nix]: Lose the `"`s.

Comment (primeos) [reply pkgs/build-support/build-fhs-userenv/env.nix]: Care to elaborate? I would keep them as they don't cause any harm (and I personally prefer quoting everything by default). While it shouldn't make a difference for Bash, there are or at least where shell where it does make a difference (IIRC).

Comment (memberbetty) [reply pkgs/build-support/build-fhs-userenv/env.nix]: Section 2.3 Token Recognition from http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_03 specifies this.

Comment (Infinisil) [reply pkgs/build-support/build-fhs-userenv/env.nix]: Doesn't really matter much, I'll merge

**Pull request merged**